src/vmm/src/cpu_config/templates.rs

```diff
@@ -195,26 +195,65 @@ where
     where
         D: Deserializer<'de>,
     {
-        let mut bitmap_str = String::deserialize(deserializer)?;
+        let original_str = String::deserialize(deserializer)?;
 
-        if bitmap_str.starts_with("0b") {
-            bitmap_str = bitmap_str[2..].to_string();
+        if !original_str.is_ascii() {
+            return Err(D::Error::custom(format!(
+                "Failed to parse string [{}] as a bitmap - non ascii characters detected",
+                original_str,
+            )));
         }
 
-        let filter_str = bitmap_str.replace('0', "1");
-        let filter_str = filter_str.replace('x', "0");
-        let value_str = bitmap_str.replace('x', "0");
+        // retain original string for potential errors
+        let mut bitmap_str = original_str.clone();
 
-        let filter = V::from_str_radix(filter_str.as_str(), 2).map_err(|err| {
+        bitmap_str.retain(|c| c != '_');
+
+        let value_bytes = if let Some(slice) = bitmap_str.strip_prefix("0b") {
+            slice.as_bytes()
+        } else {
+            bitmap_str.as_bytes()
+        };
+
+        let mut filter = value_bytes.to_owned();
+        let mut value = value_bytes.to_owned();
+
+        // replace '0' => '1'
+        filter.iter_mut().for_each(|c| {
+            if c == &b'0' {
+                *c = b'1';
+            }
+        });
+        // replace 'x' => '0'
+        filter.iter_mut().for_each(|c| {
+            if c == &b'x' {
+                *c = b'0';
+            }
+        });
+        // replace 'x' => '0'
+        value.iter_mut().for_each(|c| {
+            if c == &b'x' {
+                *c = b'0';
+            }
+        });
+
+        // # Safety:
+        // We know that vectors contain only valid ASCII characters
+        let value = unsafe { std::str::from_utf8_unchecked(&value) };
+        // # Safety:
+        // We know that vectors contain only valid ASCII characters
+        let filter = unsafe { std::str::from_utf8_unchecked(&filter) };
+
+        let filter = V::from_str_radix(filter, 2).map_err(|err| {
```

Comment:
What is the exact performance improvement from foregoing the unicode checks here? Is it quantifiable? If not, I don't see why we shouldn't use `std::str::from_utf8(&...).unwrap()` here. 

Or maybe we can operate on a char array as opposed to a byte array? The check of "is this a unicode boundary" only checks a single bit afaik, so not sure how much impact skipping that has.